### DISCUSSION TITLE
Add support to Matroska video

### DISCUSSION THREAD

**Discussion by ghost** [godotengine/godot-proposals#9474] [General Discussions]
Good morning. I would like to propose adding Matroska video format support to Godot. I am currently engaged with my project, already published: Pinbot ([www.sante76.eu](http://www.sante76.eu/)). I believe in videos as a highly expressive artistic tool. Pinbot extensively uses videos. Since it was created with Godot 3.5, I have used the WebM format, which, as you may know, is no longer supported in Godot 4. I have personally experienced that WebM is unreliable and causes total game crashes. I am preparing to switch to Theora, already with Pinbot.

However, I am encountering several issues with Theora: despite trying various settings, there is always something wrong. The game, in fact, no longer has the crash problem, but Theora videos tend to be either huge or, when attempting to reduce them, often exhibit absolutely unacceptable artifacts.

For this reason, I am here to propose the introduction of the Matroska format, the famous *.mkv videos. Matroska is an open-source format, widely used, and much more reliable than Theora, at least from a practical standpoint. Information about Matroska is available at https://matroska.org/.

Thank you.

**Comment by Calinou** [comment]
Matroska is a video *container*, not a codec. It can contain pretty much any codec for both video and audio, from Theora to H.264 to AV1.

Supporting a video container is much easier than supporting a codec, but it will not let you play videos that are encoded in a codec not supported by Godot (i.e. anything else than Theora).

The issue is that integrating video codecs is a very difficult task, as described in https://github.com/godotengine/godot-proposals/issues/3286. That's the hard part of handling video playback, and it's why the plan is still to move it to extensions instead of making it core.

**Reply by ghost** [reply]
Understood. I've taken a look at [#3286](https://github.com/godotengine/godot-proposals/issues/3286). It seems to me that there are also significant issues regarding licenses, particularly with codecs in general, especially for commercial use and finished products... Or am I mistaken?

**Reply by Calinou** [reply]
Yes, you want to stick to royalty-free codecs as much as possible: Theora, VP8, VP9 and AV1 are such examples.

H.264, H.265 (HEVC) and H.266 are all patent-encumbered in comparison.

**Comment by ghost** [comment]
> Yes, you want to stick to royalty-free codecs as much as possible: Theora, VP8, VP9 and AV1 are such examples.
> 
> H.264, H.265 (HEVC) and H.266 are all patent-encumbered in comparison.

I thank you for the clarification. Now all that's left is to wait for developments: I've seen that there's a lot of movement around the issue, and I'm confident that, in the end, you will find a solution...
